### PR TITLE
add pod securityContext clusterpolicy

### PR DESCRIPTION
**Pull request by iamvigneshwars**

### CODE REVIEW

File: charts/workflows/templates/pod-securitycontext-clusterpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      context:
+        - name: values
+          configMap:
+            name: sessionspaces
+            namespace: "{{ `{{request.object.metadata.namespace}}` }}"
+      mutate:
+        patchStrategicMerge:
+          spec:
+            securityContext:
+              runAsGroup: "{{ `{{values.data.gid | parse_json(@).to_number(@)}}` }}"
```
Comment (garryod) [inline charts/workflows/templates/pod-securitycontext-clusterpolicy.yaml:24]: I think we also want to set the `runAsUser` to a value set in the `values.yaml`. We may also want to reject any pods with `allowPrivilegeEscalation` set to `true`

File: charts/workflows/templates/pod-securitycontext-clusterpolicy.yaml
Comment (garryod) [inline charts/workflows/templates/pod-securitycontext-clusterpolicy.yaml:1]: Do we need to set `spec.containers[].securityContext` too? Or is it covered by `spec.securityContext`?

**Review by garryod — CHANGES_REQUESTED** [review]

File: charts/workflows/templates/pod-securitycontext-clusterpolicy.yaml
Comment (iamvigneshwars) [reply charts/workflows/templates/pod-securitycontext-clusterpolicy.yaml:1]: `spec.securityContext` covers it

File: charts/workflows/test-policy/pod-securitycontext/chainsaw-test.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+              image: nginx:latest
+    - assert:
+        resource:
+          apiVersion: v1
+          kind: Pod
+          metadata:
+            namespace: test
+          spec:
+            securityContext:
+              runAsGroup: 1234
```
Comment (garryod) [inline charts/workflows/test-policy/pod-securitycontext/chainsaw-test.yaml:46]: We also want to test the `runAsUser` is set appropriately and that setting `allowPrivilegeEscalation` results in a rejection

**Review by garryod — CHANGES_REQUESTED** [review]

**Review by garryod — APPROVED** [review]

**Pull request merged**